openedx/core/djangoapps/schedules/management/commands/tests/send_email_base.py

```diff
@@ -344,10 +349,8 @@ def _assert_template_for_offset(self, offset, message_count):
 
         user = UserFactory.create()
         for course_index in range(message_count):
-            ScheduleFactory.create(
-                start=target_day,
-                upgrade_deadline=upgrade_deadline,
-                enrollment__course__self_paced=True,
+            self._schedule_factory(
+                offset=offset,
```

Comment:
yeah - seeing how we need to pass in `offset` to both `self._get_dates` in this method as well as `self._schedule_factory` so it can be passed into the same method again, it makes sense to have `self._schedule_factory` not call `self._schedule_factory` itself.

Reply:
I actually prefer the way I have it now. Those parameters would need to be passed in to *every* call to `_schedule_factory`. The way it is now, I can just call `self._schedule_factory()` and it creates a useful schedule.